diff --git a/node_modules/react-native-device-info/RNDeviceInfo.podspec b/node_modules/react-native-device-info/RNDeviceInfo.podspec
index 78f76c5..97ce95d 100644
--- a/node_modules/react-native-device-info/RNDeviceInfo.podspec
+++ b/node_modules/react-native-device-info/RNDeviceInfo.podspec
@@ -13,5 +13,6 @@ Pod::Spec.new do |s|
   s.source       = { :git => "https://github.com/rebeccahughes/react-native-device-info.git" }
 
   s.source_files  = "RNDeviceInfo/*.{h,m}"
+  s.dependency 'React-Core'
 
 end
